### ISSUE TITLE
Modus Table - lazy loading

### ISSUE THREAD

**Issue by Akashmuthumanickam** [trimble-oss/modus-web-components#2522]
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [Modus style guidelines](https://modus.trimble.com/components/web/introduction/)

### Proposal

Need a lazy loading concept on Modus table. It should be server side and it keeps tracking the page index and count. 

### What is the issue regarding ?

@trimble-oss/modus-web-components

### Priority

High

### What product/project are you using Modus Components for ?

Flux - Connect web extensions

### What is your team/division name ?

CPD

### Are you willing to contribute ?

Yes

**Comment by enowak1031** [comment]
#1442

**Comment by prashanth-offcl** [comment]
@Akashmuthumanickam The issue title and description convey different things. The title refers to Virtualization but based on the description I think you want us to support infinite scrolling by providing you with an option with which you can call the API to get the next page data and feed us with the new set. Can you please clarify your question?

**Comment by Akashmuthumanickam** [comment]
@prashanth-offcl Yes the main goal of this ticket is to implement lazy loading. Apologies for the title,
